client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,171 @@
+			hideRecommendationLabel={ true }
+		/>
+	);
+}
+
+function MarketplaceShoppingCart( { onAddDomainToCart, selectedDomain } ) {
+	const { responseCart, isLoading } = useShoppingCart();
+	const { products, sub_total_display } = responseCart;
+
+	return isLoading ? (
+		<div></div>
+	) : (
+		<>
+			<h1 className="marketplace-domain-upsell__shopping-cart marketplace-title">Your cart</h1>
+			<div className="marketplace-domain-upsell__shopping-cart basket-items">
+				{ products.map( ( product ) => {
+					return (
+						<div
+							key={ product.product_name }
+							className="marketplace-domain-upsell__shopping-cart basket-item"
+						>
+							<div>{ product.product_name }</div>
+							<div>{ product.item_subtotal_display }</div>
+						</div>
+					);
+				} ) }
+			</div>
+			<hr />
+			<h1 className="marketplace-domain-upsell__shopping-cart total">
+				<div> Total </div>
+				<div>{ sub_total_display }</div>
+			</h1>
+
+			<Button
+				onClick={ onAddDomainToCart }
+				isBusy={ false }
+				isPrimary
+				disabled={ selectedDomain === null }
+			>
+				Checkout
+			</Button>
+		</>
+	);
+}
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomain, setDomain ] = useState( null );
+	const { addProductsToCart, replaceProductsInCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ { ...yoastProduct } ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const onAddDomainToCart = () => {
+		const { product_slug, domain_name } = selectedDomain;
+		const domainProduct = {
+			...domainRegistration( {
+				productSlug: product_slug,
+				domain: domain_name,
+				source: 'Marketplace-Yoast-Domain-Upsell',
+			} ),
+			...selectedDomain,
+		};
+		addProductsToCart( [ domainProduct ] );
+		window.location.href = window.location.origin + '/checkout/' + selectedSite.slug;
+	};
+
+	const onDomainSelect = ( suggestion: DomainSuggestion ): void => {
+		setDomain( suggestion );
+	};
+
+	return (
+		<div className="marketplace-domain-upsell__root">
+			<div className="marketplace-domain-upsell__back-button">
+				<Button isBusy={ false } isLink>
+					<svg
+						width="17"
+						height="13"
+						viewBox="0 0 17 13"
+						fill="none"
+						xmlns="http://www.w3.org/2000/svg"
+					>
+						<path
+							d="M2 6.49999L7.25 0.749995M2 6.49999L17 6.5M2 6.49999L7.25 11.75"
+							stroke="#1D2327"
+							strokeWidth="1.5"
+						/>
+					</svg>
```

Comment:
Let's use the standardized `arrow-left` gridicon here instead of drawing your own arrow.

Reply:
We could also try using Gutenberg's [dashicons](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/icon#icon) - last time I checked there were 3 possible arrow variations we could use here.

Reply:
Fixed by 0aa07864d5e9a9480c7959ac92405d761970af1e I used @wordpress/icons